lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1984,7 +1984,12 @@ def init_api_client
         end
         port = ":#{uri.port}" if uri.port
         @client = Google::Cloud::Logging::V2::LoggingServiceV2Client.new(
-          credentials: GRPC::Core::Channel.new("#{host}#{port}", nil, creds))
+          credentials: GRPC::Core::Channel.new(
+            "#{host}#{port}", {
+              'grpc.primary_user_agent' =>
+                "#{PLUGIN_NAME}/#{PLUGIN_VERSION} grpc-ruby/#{GRPC::VERSION} " \
```

Comment:
"grpc-ruby"? Do we mean to include a Ruby version? What's the "-ruby" for?

Reply:
Oh, I guess it's the gRPC Ruby library (version)?

Reply:
Yes, this is simulating what the REST libraries set the User Agent to — they concatenate `application_name`/`application_version` with the default User Agent string (`google-cloud-ruby/#{VERSION}`). I'm doing the same in the gRPC case.

Reply:
Hmm... I wonder why they chose `google-cloud-ruby` over `google-api-ruby-client`. Because the gRPC library is called [google-cloud-ruby](https://github.com/GoogleCloudPlatform/google-cloud-ruby). While the non-grpc library is called [google-api-ruby-client](https://github.com/google/google-api-ruby-client).

Since this is the gRPC path we are adding, we should call it `google-cloud-ruby`. But then the GRPC::VERSION is not the actual `google-cloud-ruby` version, which is still a bit confusing.

Reply:
You're right, I got confused. They use `google-api-ruby-client` ([source](https://github.com/google/google-api-ruby-client/blob/master/lib/google/apis/core/base_service.rb#L431)). I do think we should append the default User Agent used by the underlying library, though, which is actually `grpc-ruby/#{VERSION}`.

Reply:
Is this the implementation in the REST path? https://github.com/google/google-api-ruby-client/blob/24be95a758ed3c4e559b8e31fd20ad32d166bd18/lib/google/apis/core/base_service.rb#L431

```
        # Build the user agent header
        # @return [String]
        def user_agent
          sprintf('%s/%s google-api-ruby-client/%s %s (gzip)',
                  client_options.application_name,
                  client_options.application_version,
                  Google::Apis::VERSION,
                  Google::Apis::OS_VERSION)
        end
```

Seems they are using `google-api-ruby-client` with the API version.

Reply:
Our messages crossed. We both pointed to the same code. :-)

Reply:
This is what the current code sends as the value of the User-Agent: `Fluentd Google Cloud Logging plugin/0.6.23 grpc-ruby/1.8.3`. Verified on a round-trip.